### ISSUE TITLE
`jsx-no-undef` seems to ignore `allowGlobals` option, plus unclear examples

### ISSUE THREAD

**Issue by c-vetter** [jsx-eslint/eslint-plugin-react#3211]
Firstly, I find the docs page examples difficult to understand. The way I interpret it, `allowGlobals` should specify whether to include or ignore the `globals` setting from the `eslintrc`. If that is correct, the option seems to be ignored. Otherwise, please clarify the intended behavior.

----

Given this file:
```jsx
/* eslint { "react/jsx-no-undef": ["error", { allowGlobals: false }] } */
var Hello = React.createClass({
  render: function() {
    return <Text>Hello</Text>;
  }
});
module.exports = Hello;
```
I get this error:
```log
  4:13  error  'Text' is not defined  react/jsx-no-undef
```
If I add `Text` the `globals` setting, the error disappears even though I leave `allowGlobals` `false`.

Using `eslint-plugin-react` v7.28.0

**Comment by ljharb** [comment]
That's working as expected. `allowGlobals` just means the global scope is considered.

The tests have, for allowGlobals false:
```jsx
        const TextWrapper = function (props) {
          return (
            <Text />
          );
        };
        export default TextWrapper;
```
as invalid, and:
```jsx
        import Text from "cool-module";
        const TextWrapper = function (props) {
          return (
            <Text />
          );
        };
```
as valid.

**Comment by nyngwang** [comment]
@c-vetter I cannot reproduce your problem. The error did show. I have `globals` set and `"allowGlobals": false`. (I use `.eslintrc.json` btw)

<img src="https://user-images.githubusercontent.com/24765272/154839016-6be41f8d-f529-4398-8b1f-227de917b92d.png" width="700">
